Sources/Core/EthereumABI/ABIElements.swift

```diff
@@ -325,7 +325,7 @@ extension ABI.Element.Function {
             // set a flag to detect the request succeeded
         }
 
-        if returnArray.isEmpty {
+        if returnArray.isEmpty && !outputs.isEmpty {
```

Comment:
I know I'm annoying with this but can we please make a test here. 😄

Reply:
Agree.

Reply:
I will work on this

Reply:
Just wonder, does it resolved yet, @pharms-eth?